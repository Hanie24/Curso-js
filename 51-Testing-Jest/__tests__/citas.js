import Citas from "../js/classes/Citas";

describe("Probar la Clase de citas", () => {
  const citas = new Citas();

  const id = Date.now();

  test("Agregar una nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Foo",
      propietario: "Emmanuelle",
      telefono: "12344567",
      fecha: "10/10/2024",
      hora: "11:11",
      sintomas: "nada",
    };

    citas.agregarCita(citaObj);

    // Pueba
    expect(citas).toMatchSnapshot();
  });

  test("actualizar cita", () => {
    const citaActualizada = {
      id,
      mascota: "Simba",
      propietario: "Hanie",
      telefono: "12344567",
      fecha: "10/10/2024",
      hora: "11:11",
      sintomas: "nada",
    };

    citas.editarCita(citaActualizada);

    expect(citas).toMatchSnapshot();
  });

  test("Eliminar cita", () => {
    citas.eliminarCita(id);

    expect(citas).toMatchSnapshot();
  });
});
