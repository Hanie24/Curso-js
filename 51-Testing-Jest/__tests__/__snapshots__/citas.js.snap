// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la Clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10/10/2024",
      "hora": "11:11",
      "id": 1731439859646,
      "mascota": "Foo",
      "propietario": "Emmanuelle",
      "sintomas": "nada",
      "telefono": "12344567",
    },
  ],
}
`;

exports[`Probar la Clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Probar la Clase de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10/10/2024",
      "hora": "11:11",
      "id": 1731439859646,
      "mascota": "Simba",
      "propietario": "Hanie",
      "sintomas": "nada",
      "telefono": "12344567",
    },
  ],
}
`;
